modules/assim.batch/R/pda.neff.R

```diff
@@ -0,0 +1,163 @@
+##' Autocorelation correction and efficient sample size calculation on latent process
+##'
+##' @title Calculate N_eff 
+##' @param inputs list
+##'
+##' @return inputs list, updated inputs with n_eff
+##'
+##' @author Istem Fer
+##' @export
+pda.neff.calc <- function(inputs){
+  
+  
+  for(i in seq_along(inputs)){
+    
+    n   <- inputs[[i]]$n
+    # for now we're doing autocorrelation correction on flux data only
+    # NEE, LE, FC
+    if(inputs[[i]]$variable.id %in% c(297, 298, 1000000042)){
+      
+      rho      <- pda.autocorr.calc(inputs[[i]], "heteroskedastic.laplacian")
+      n_eff    <- n*(1-rho)/(1+rho)
+      inputs[[i]]$n_eff <- n_eff
+      
+    }else{
+      # assume n_eff is the as n
+      inputs[[i]]$n_eff <- n
+    }
+  }
+  
+  
+} # pda.neff.calc
+
+##' @title autocorrelation correction
+##' @param input list that contains time-series data vector and parameters for heteroskedastic.laplacian
+##' @param model data model type, for flux data heteroskedastic laplacian, normal is an example 
+##'
+##' @return rho AR(1)
+##'
+##' @author Istem Fer
+##' @export
+pda.autocorr.calc <- function(input, model = "heteroskedastic.laplacian"){
+  
+  library(rjags)
```

Comment:
`library` calls inside of functions are not a good idea. Since you only need like 2 functions from this package, I would just add `rjags::<name>` to them.

Also, you may be able to simplify some of the running syntax with the `runjags` package, which also has the added bonus of being easier to install _and_ can automatically run until convergence, which is better than arbitrarily setting `niter` as you do now.